agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/exporter/UrlParser.java

```diff
@@ -21,16 +21,68 @@
 
 package com.microsoft.applicationinsights.agent.internal.exporter;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
 class UrlParser {
 
   /**
    * Returns the path portion of the url.
    *
    * <p>Returns {@code null} if the path cannot be extracted from url for any reason.
    */
+  @Nullable
   static String getPathFromUrl(String url) {
```

Comment:
It looks like URL.getPath does the same thing? it handles case like ?query and #. Can we convert url string to an URL object and then use getPath directly, and we can get rid of UrlParser?

Reply:
constructing URL objects is slow: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3699

Reply:
it will be nice to compare the perf side by side. was URLStreamHandler.parseURL slow or the constructor of URL? 
inside the constructor it will invoke parseURL from the handler.. I am wondering if the newer version of java has improved it..